marlowe-dashboard-client/src/Dashboard/View.purs

```diff
@@ -431,33 +431,35 @@ currentWalletCard walletDetails =
 
     assets = view _assets walletDetails
   in
-    div [ classNames [ "p-5", "pb-6", "md:pb-8" ] ]
-      [ h3
-          [ classNames [ "font-semibold", "mb-4", "truncate", "w-11/12" ] ]
+    div [ classNames [ "h-full", "grid", "grid-rows-auto-1fr-auto" ] ]
+      [ h2
+          [ classNames Css.cardHeader ]
           [ text $ "Wallet " <> walletNickname ]
-      , div
-          [ classNames Css.hasNestedLabel ]
-          [ label
-              [ classNames Css.nestedLabel ]
-              [ text "Wallet ID" ]
-          , input
-              [ type_ InputText
-              , classNames $ Css.input true <> [ "mb-4" ]
-              , value $ UUID.toString $ unwrap companionAppId
-              , readOnly true
+      , div [ classNames [ "p-4", "h-full", "flex", "flex-col" ] ]
```

Comment:
This is more complicated than it needs to be. The grid of the parent already takes care of this sizing for you. Also we should set vertical overflow - unlikely to be needed when there's so little in the div, but you never know, it might get added to in the future.

```suggestion
      , div [ classNames [ "p-4", "overflow-y-auto" ] ]
```